File: Algorithm/SWEA_8191_만화책정렬하기.cpp
#include <iostream>
using namespace std;

#define MAX 200000

int main(void) {
    ios::sync_with_stdio(0);
    cin.tie(0);
    cout.tie(0);

    int T;
    cin >> T;
    for (int i = 1; i < T + 1; i++) {
        int N;
        cin >> N;
        int *books = new int[N];
        int tmp_list[MAX][MAX];
        
        for (int j = 0; j < N; j++) {
            int isTrue = -1;
            for (int t; t < sizeof(tmp_list); t++) {
                if (tmp_list[t][-1] + 1 == books[j]) {
                    isTrue = t;
                }
            }
        }
    }
}

File: Algorithm/SWEA_8191_만화책정렬하기.py
T = int(input())
for i in range(1, T + 1):
    N = int(input())
    books = map(int, input().split())
    tmp_list = []
    for book in books:
        # 리스트의 마지막 요소 + 1 이 book과 같은 것이 있다면 그 리스트에 book 추가하기
        # 없다면 새로운 리스트 생성후 book 추가하기
        isTrue = -1
        for t in range(len(tmp_list)):
            if tmp_list[t][-1] + 1 == book:
                isTrue = t
        if isTrue != -1:
            tmp_list[isTrue].append(book)
        else:
            tmp_list.append([book])

    print(f"#{i} {len(tmp_list)}")
